### PR TITLE
Cache RabbitMQ connections and HTTP Sessions in Octopoes

### PR DESCRIPTION
**Pull request by Donnype**
### PLEASE QA THIS PROPERLY, I HAVE SEEN SOME ERRORS DUE TO RABBIT GOING DOWN IN THE LOGS BUT I AM NOT SURE IF THIS WAS DUE TO STARTUP RACE CONDITIONS.

- Cache the octopoes http session and use one rabbitmq connection per celery worker, and one rabbitmq connection for the api (one process).
- Use native support from FastAPI and Celery to spawn connections on startup of the worker/api. 
- Use native support from FastAPI and Celery to close RabbitMQ connections on shutdown of the worker/api. 

Some considerations and issues with the current approach:
- [Pika does not claim that the blocking connections are thread safe](https://pika.readthedocs.io/en/latest/faq.html), but the other services such as the scheduler has not experienced issues with this. For now I have added a `threading.local()` instance to resolve this.
- Running uvicorn with the `--reload` flag means that if we cannot connect to RabbitMQ on application startup (reasonable scenario when we rebuild kat for instance, or when we temporarily kill rabbitmq), the container does not exit. Else the container would kill itself when you code a syntax error during development. But this means that we have to build in retry logic for the initial connection ourselves, so I added RabbitMQ as a dependency that has to be healthy in the docker-compose file. There are still edge cases here where you start octopoes_api with `--no-deps` and RabbitMQ is down. You have to then manually restart Octopoes API since there is no retry logic at the application level implemented. A code change would however be detected by Uvicorn and restart the api successfully in that scenario. We could:
  * Ignore the exception and start the application anyway: then we might see the exception re-appear but at least the application is started so we would stay live and try creating connection until it succeeds [current solution]
  * Build in (longer) retry logic at startup as a lot of Octopoes is not really usable without RabbitMQ
  * (Interesting and ugly thing that I found out also works: if we catch the startup-connection-exception and kill the uvicorn process using `os.kill(os.getppid())`, the container actually properly dies and is restarted by docker due to our `restart: unless-stopped` policy! :nerd_face: )

Celery seems to have proper logic to respawn the process when we cannot connect workers on `worker.init`.

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/1162

### Celery workers

Celery workers connecting on startup:

celerycon

Celery workers disconnecting on shutdown:

celdis

---

### FastAPI

FastAPI connecting on startup:

rabbitcon

FastAPI disconnecting on shutdown:

rabbitclose

FastAPI failing to connect but starting anyways:

image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — APPROVED** [review]

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* No apparent regressions
* Restarting containers sequentially one-by-one results in a working deployment

**Comment by Donnype** [comment]
I added `pytest-timeout` to fix at least the sporadic hanging of the integration test.

File: octopoes/octopoes/api/router.py [hunk 1]
```diff
@@ -76,21 +76,24 @@ def settings() -> Settings:
...
+def xtdb_session(
+    client: str = Depends(extract_client), settings_: Settings = Depends(settings)
+) -> Generator[XTDBSession, None, None]:
+    session = XTDBSession(get_xtdb_client(settings_.xtdb_uri, client, settings_.xtdb_type))
+
+    try:
+        yield session
+    finally:
+        session.commit()
+        logger.info("Committed XTDBSession")
```
Comment (dekkers) [inline octopoes/octopoes/api/router.py:88]: The old code also did not commit when an exception happened and I don't think that we want that, so we shouldn't do that inside finally.

**Review by dekkers — CHANGES_REQUESTED** [review]

**Comment by Darwinkel** [comment]
Re-tested. Still works fine! :)

**Review by dekkers — APPROVED** [review]

**Pull request merged**